application/models/post_test.go

```diff
@@ -1102,6 +1102,218 @@ func (ms *ModelSuite) TestPost_GetProvider() {
+				{Status: PostStatusOpen, IsBackStep: true},
+				{Status: PostStatusReceived},
+				{Status: PostStatusCompleted},
+				{Status: PostStatusRemoved},
+			},
+		},
+		{name: "Delivered Post - Creator",
+			post: Post{ID: 1, CreatedByID: 11, Status: PostStatusDelivered},
+			user: User{ID: 11},
+			want: []StatusTransitionTarget{{Status: PostStatusCompleted}},
+		},
+		{name: "Completed Post - Creator",
+			post: Post{ID: 1, CreatedByID: 11, Status: PostStatusCompleted},
+			user: User{ID: 11},
+			want: []StatusTransitionTarget{
+				{Status: PostStatusAccepted, IsBackStep: true},
+				{Status: PostStatusDelivered, IsBackStep: true},
+			},
+		},
+		{name: "Accepted Post - Provider",
+			post: Post{ID: 1, ProviderID: nulls.NewInt(12), Status: PostStatusAccepted},
+			user: User{ID: 12},
+			want: []StatusTransitionTarget{
+				{Status: PostStatusDelivered, isProviderAction: true},
+			},
+		},
+		{name: "Delivered Post - Provider",
+			post: Post{ID: 1, ProviderID: nulls.NewInt(12), Status: PostStatusDelivered},
+			user: User{ID: 12},
+			want: []StatusTransitionTarget{
+				{Status: PostStatusAccepted, IsBackStep: true, isProviderAction: true},
+			},
+		},
+		{name: "Completed Post - Provider",
+			post: Post{ID: 1, ProviderID: nulls.NewInt(12), Status: PostStatusCompleted},
+			user: User{ID: 12},
+			want: []StatusTransitionTarget{},
+		},
+		{name: "Open Post - Not Creator Or Provider",
+			post: Post{ID: 1, Status: PostStatusOpen},
+			user: User{ID: 99}, want: []StatusTransitionTarget{},
+		},
+		{name: "Accepted Post - Not Creator Or Provider",
+			post: Post{ID: 1, Status: PostStatusAccepted},
+			user: User{ID: 99}, want: []StatusTransitionTarget{},
+		},
+	}
+
+	for _, tt := range tests {
+		ms.T().Run(tt.name, func(t *testing.T) {
+			got, err := tt.post.GetStatusTransitions(tt.user)
+			ms.NoError(err)
+			ms.Equal(tt.want, got, "incorrect status transitions")
+		})
+	}
+}
+
+func (ms *ModelSuite) TestPost_GetPotentialProviderActions() {
+	f := createUserFixtures(ms.DB, 3)
+	users := f.Users
+	posts := createPostFixtures(ms.DB, 2, false)
+	createPotentialProviderFixtures(ms.DB, 0, 2)
+
+	acceptedPost := posts[0]
+	acceptedPost.Status = PostStatusAccepted // This doesn't change the post in the slice
+
+	tests := []struct {
+		name string
+		post Post
+		user User
+		want []string
+	}{
+		{name: "Open Post - Creator",
+			post: posts[1],
+			user: users[0],
+			want: []string{},
+		},
+		{name: "Open Post with no offers - not Creator",
+			post: posts[1],
+			user: users[1],
+			want: []string{RequestActionOffer},
+		},
+		{name: "Open Post with offer - Offerer",
+			post: posts[0],
+			user: users[1],
+			want: []string{RequestActionRetractOffer},
+		},
+	}
+	for _, tt := range tests {
+		ms.T().Run(tt.name, func(t *testing.T) {
+			got, err := tt.post.GetPotentialProviderActions(tt.user)
+			ms.NoError(err)
+			ms.Equal(tt.want, got, "incorrect actions")
+		})
+	}
+}
+
+func (ms *ModelSuite) TestPost_GetCurrentActions() {
```

Comment:
This test could be just a subset of the possible input combinations since the lower-level functions have already tested all the permutations. Kind of like you did in the gql test.

Reply:
:+1: